builtin/logical/postgresql/secret_creds.go

```diff
@@ -91,12 +91,25 @@ func (b *backend) secretCredsRevoke(
 	}
 	username, ok := usernameRaw.(string)
 
+
 	// Get our connection
 	db, err := b.DB(req.Storage)
 	if err != nil {
 		return nil, err
 	}
 
+	// Check if the role exists
+	var exists bool
+	query := fmt.Sprintf("SELECT exists (SELECT rolname FROM pg_roles WHERE rolname='%s');", username);
+	err = db.QueryRow(query).Scan(&exists)
+	if err != nil && err != sql.ErrNoRows {
+		return nil, err
+	}
+
+	if exists == false {
+		return nil, nil
+	}
```

Comment:
Ideally this should not be added. However i could not get acceptance tests to pass otherwise (even on master).
A few tests do revocation as part of the test - But the test cleanup also does revocation, which results in a lot of failed queries because the role in PG no longer exists.

I'm not sure if this should be handled by some sort of secret lookup.